src/Pages/ProductDetails/ProductDetails.js

```diff
@@ -0,0 +1,108 @@
+import { useParams } from "react-router-dom";
+import { useData } from "../../Contexts";
+import { Link, useNavigate } from "react-router-dom";
+import "./ProductDetails.css";
+
+export function ProductDetails() {
+  const navigate = useNavigate();
+  const { productId } = useParams();
+  const { data, itemsInCart, itemsInWishList, dispatch } = useData();
+  const product = data.find((item) => item.id === productId);
+  const {
+    id,
+    name,
+    image,
+    price,
+    material,
+    brand,
+    inStock,
+    fastDelivery,
+    ratings,
+    offer,
+    idealFor,
+    level,
+    color,
+  } = product;
+
+  const CartButtons = () => {
```

Comment:
hey rohit, i see you've used `CartButtons()` and `WishlistButtons()` component many times in your files, you can separately write them in a file and then import wherever you wish to use them, it will increase reusability of the code.